src-tauri/src/main.rs

```diff
@@ -1,80 +1,116 @@
+}
+
 mod aux_functions;
 use aux_functions::commands::{run_articulation_point, run_global_mincut, run_stoer_wagner};
-use tauri::Manager;
-use tokio::sync::{mpsc, Mutex};
-use tracing::info;
 use tracing_subscriber;
-struct AsyncProcInputTx {
-    inner: Mutex<mpsc::Sender<String>>,
-}
 
 fn main() {
     tracing_subscriber::fmt::init();
 
-    let (async_proc_input_tx, async_proc_input_rx) = mpsc::channel(1); // Inputs to async process (from JS, to worker thread)
-    let (async_proc_output_tx, mut async_proc_output_rx) = mpsc::channel(1); // Outputs from async process (from worker thread, to JS)
-
     tauri::Builder::default()
-        .manage(AsyncProcInputTx {
-            inner: Mutex::new(async_proc_input_tx),
+        .manage(ActiveSerialConnection {
+            inner: Arc::new(Mutex::new(None)),
         })
         .invoke_handler(tauri::generate_handler![
-            js2rs,
             run_articulation_point,
             run_global_mincut,
             run_stoer_wagner
         ])
-        .setup(|app| {
-            tauri::async_runtime::spawn(async move {
-                async_process_model(async_proc_input_rx, async_proc_output_tx).await
-            });
-
-            let app_handle_echo = app.handle().clone();
-
-            tauri::async_runtime::spawn(async move {
-                loop {
-                    if let Some(output) = async_proc_output_rx.recv().await {
-                        rs2js(output, &app_handle_echo);
-                    }
-                }
-            });
-
-            Ok(())
-        })
+        .invoke_handler(tauri::generate_handler![
+            get_all_serial_ports,
+            connect_to_serial_port,
+            send_text
+        ])
         .run(tauri::generate_context!())
         .expect("error while running tauri application");
 }
 
-fn rs2js<R: tauri::Runtime>(message: String, manager: &impl Manager<R>) {
-    info!(?message, "rs2js");
-    manager
-        .emit_all("rs2js", format!("rs: {}", message))
-        .unwrap();
+#[tauri::command]
+fn get_all_serial_ports() -> Vec<String> {
+    SerialConnection::get_available_ports()
 }
 
 #[tauri::command]
-async fn js2rs(message: String, state: tauri::State<'_, AsyncProcInputTx>) -> Result<(), String> {
-    info!(?message, "js2rs");
-    let async_proc_input_tx = state.inner.lock().await;
-
-    async_proc_input_tx
-        .send(message)
-        .await
-        .map_err(|e| e.to_string())
-}
+fn connect_to_serial_port(
+    port_name: String,
+    app_handle: tauri::AppHandle,
+    state: tauri::State<'_, ActiveSerialConnection>,
+) -> bool {
+    let mut connection: SerialConnection = MeshConnection::new();
+
+    match connection.connect(port_name, 115_200) {
+        Ok(_h) => (),
+        Err(_e) => {
+            eprintln!("Could not connect to radio");
+            return false;
+        }
+    };
+
+    let mut decoded_listener = match connection.on_decoded_packet.as_ref() {
+        Some(l) => l.resubscribe(),
```

Comment:
do these variable names represent anything or are they random?

Reply:
Oh okay you are using the first letter

Reply:
Yeah, happy to change that if it doesn't make sense or isn't clear

Reply:
all good!